tools/screenshot-test/src/app/firebase.service.ts

```diff
@@ -0,0 +1,204 @@
+  result: boolean;
+  /** Test results: passed or failed for each test */
+  results: Map<string, boolean> = new Map<string, boolean>();
+  /**
+   * Test approved: whether the test images are copied to goldens.
+   * The value is the commit SHA and date/time of approval.
+   */
+  approved: string;
+
+  githubStatus: any;
+
+  /** Viewing mode, can be flip, diff, side */
+  mode: 'diff' | 'side' | 'flip' = 'diff';
+  /** Viewing flipping, whether the image is test image or golden image */
+  flipping: boolean = false;
+  /** Viewing collapsed: whether the result should be collapsed/expanded */
+  collapse: boolean[];
+
+  setCollapse(value: boolean) {
+    if (this.collapse) {
+      for (let i = 0; i < this.collapse.length; i++) {
+        this.collapse[i] = value;
+      }
+    }
+  }
+
+  get prLink() {
+    return `https://github.com/${config.repoSlug}/pull/${this.prNumber}`;
+  }
+
+  get commitLink() {
+    return `https://github.com/${config.repoSlug}/commit/${this.sha}`;
+  }
+
+  get travisLink() {
+    return `https://travis-ci.org/${config.repoSlug}/jobs/${this.travis}`;
+  }
+}
+
+
+@Injectable()
+export class FirebaseService {
+
+  /** The current user */
+  user: any;
+
+  /** The screenshot results */
+  screenshotResult: ScreenshotResult;
+
+  constructor() {
+    // Initialize Firebase
+    firebase.initializeApp(config.firebase);
+
+    firebase.auth().onAuthStateChanged((user) => {
+      this.user = user;
+    });
+  }
+
+
+  /** Get the firebase storage test image folder ref */
+  testRef(): firebase.storage.Reference {
+    return this._storageRef().child('test');
+  }
+
+  /** Get the firebase storage diff image folder ref */
+  diffRef(): firebase.storage.Reference {
+    return this._storageRef().child('diff');
+  }
+
+  /** Get the firebase storage golden image folder ref */
+  goldRef(): firebase.storage.Reference {
+    return firebase.storage().ref('goldens');
+  }
+
+  /** Set pull request number. All test information and pull request information will be retrived
+   * from database
+   */
+  set prNumber(prNumber: string){
+    this.screenshotResult = new ScreenshotResult();
+    this.screenshotResult.prNumber = prNumber;
+    if (!prNumber) {
+      return;
+    }
+
+    this._databaseRef().on('value', (snapshot: firebase.database.DataSnapshot) => {
+      let counter = 0;
+      snapshot.forEach((childSnapshot: firebase.database.DataSnapshot) => {
+        let childValue = childSnapshot.val();
+        switch (childSnapshot.key) {
+          case 'sha':
+            this._processSha(childValue);
+            break;
+          case 'travis':
+            this.screenshotResult.travis = childValue;
+            break;
+          case 'results':
+            this._processResults(childSnapshot);
+            break;
+          case 'approved':
+            this.screenshotResult.approved = childValue;
+            break;
+        }
+        counter ++;
```

Comment:
Just for the future: There shouldn't be any space between the `++` . Also the `if` below. Strict checking is really recommended.